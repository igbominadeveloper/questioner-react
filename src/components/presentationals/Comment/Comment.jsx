import React from 'react';
import moment from 'moment';

const Comment = ({ comment }) => {
  return (
    <div className="single-comment flex p-1">
      <div className="avatar bg-black">
        <p>
          {comment.user.firstname.charAt(0).toUpperCase()}
          {comment.user.lastname.charAt(0).toUpperCase()}
        </p>
      </div>
      <div className="comment-details">
        <div className="comment-content">
          <div className="upper-details flex space-between">
            <p className="comment-user">
              <a href="#" className="text-black text-bold">
                {comment.user.firstname}&nbsp;
                {comment.user.lastname}
              </a>
            </p>
            <p className="text-light comment-time">
              <i className="fa fa-clock-o" />
              {moment(comment.created_at).fromNow()}
            </p>
          </div>
          <p className="text-light mt-1">{comment.comment}</p>
        </div>
      </div>
    </div>
  );
};

export default Comment;
