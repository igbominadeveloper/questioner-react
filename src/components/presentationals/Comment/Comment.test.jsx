import React from 'react';
import { shallow } from 'enzyme';

import Comment from './Comment';
const props = {
  comment: {
    comment: '',
    user: {
      firstname: '',
      lastname: '',
    },
    created_at: Date.now(),
  },
};

const component = shallow(<Comment {...props} />);

describe('Comment', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
