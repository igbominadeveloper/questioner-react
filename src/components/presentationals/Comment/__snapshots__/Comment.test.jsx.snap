// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders without errors 1`] = `
<div
  className="single-comment flex p-1"
>
  <div
    className="avatar bg-black"
  >
    <p />
  </div>
  <div
    className="comment-details"
  >
    <div
      className="comment-content"
    >
      <div
        className="upper-details flex space-between"
      >
        <p
          className="comment-user"
        >
          <a
            className="text-black text-bold"
            href="#"
          >
            Â 
          </a>
        </p>
        <p
          className="text-light comment-time"
        >
          <i
            className="fa fa-clock-o"
          />
          a few seconds ago
        </p>
      </div>
      <p
        className="text-light mt-1"
      />
    </div>
  </div>
</div>
`;
